app/src/pages/playground/PlaygroundChatTemplate.tsx

```diff
@@ -183,6 +202,95 @@ export function PlaygroundChatTemplate(props: PlaygroundChatTemplateProps) {
   );
 }
 
+function MessageEditor({
+  message,
+  updateMessage,
+  templateLanguage,
+  playgroundInstanceId,
+  template,
+  messageMode,
+}: {
+  playgroundInstanceId: number;
+  message: ChatMessage;
+  template: PlaygroundChatTemplateType;
+  templateLanguage: TemplateLanguage;
+  updateMessage: (patch: Partial<ChatMessage>) => void;
+  messageMode: MessageMode;
+}) {
+  if (messageMode === "toolCalls") {
+    return (
+      <View
+        paddingTop="size-100"
+        paddingStart="size-200"
+        paddingEnd="size-200"
+        paddingBottom="size-200"
+      >
+        <Field label={"Tool Calls"}>
+          <CodeWrap width={"100%"}>
+            <ChatMessageToolCallsEditor
+              playgroundInstanceId={playgroundInstanceId}
+              toolCalls={message.toolCalls}
+              templateMessages={template.messages}
+              messageId={message.id}
+            />
+          </CodeWrap>
+        </Field>
+      </View>
+    );
+  }
+  if (message.role === "tool") {
+    return (
+      <Form
+        onSubmit={() => {
+          //Do nothing
```

Comment:
Should we prevent default or does our Form component already do that?

Reply:
oh no i think i need to do that, i think this was just copy pasta

Reply:
added prevent default i think technically this will work but prevent default is more clear!